### PR TITLE
closes #85

### LINKED ISSUE #85: Hiding sidebar when selectedBlock is not empty
Hello,

I think sidebar should be hidden when the selected block has text.
I can create a PR with this feature if it`s ok.

Thanks

### PR DESCRIPTION
**Pull request by SeverS**

### CODE REVIEW

File: tests/components/MegadraftEditor_test.js [hunk 1]
```diff
@@ -204,4 +204,43 @@ describe("MegadraftEditor Component", () => {
...
     );
   });
+
+  it("renders default sidebar if sidebarRendererFn not provided", function() {
+    const sidebar = this.wrapper.find(Sidebar);
+    expect(sidebar).to.have.length(1);
+  });
+
+  it("calls sidebarRendererFn if it's provided", function() {
+    const rcs = sinon.spy();
```
Comment (ricobl) [inline tests/components/MegadraftEditor_test.js]: Is `rcs` something like `renderCustomSidebar`? Can we make it more obvious?

File: src/components/MegadraftEditor.js [hunk 2]
```diff
@@ -152,6 +154,23 @@ export default class MegadraftEditor extends Component {
     }
   }
 
+  getSidebarComponent = () => {
+    const { sidebarRendererFn, editorState } = this.props;
+    let SidebarComponent = null;
+    if(typeof sidebarRendererFn === "function") {
+      SidebarComponent = sidebarRendererFn(this.plugins, editorState);
```
Comment (SeverS) [inline src/components/MegadraftEditor.js]: onChange will be passed from the upper level component. I have made an example in docs. 

Good catch about `return` statement. I'll do it.

File: src/components/MegadraftEditor.js [hunk 3]
```diff
@@ -152,6 +154,23 @@ export default class MegadraftEditor extends Component {
     }
   }
 
+  getSidebarComponent = () => {
```
Comment (ricobl) [inline src/components/MegadraftEditor.js]: That's a good point but I believe it won't be a problem, we can control the props from the `render` method.

File: src/components/MegadraftEditor.js (hunk 2 above)
Comment (ricobl) [inline src/components/MegadraftEditor.js]: Yeah, that was great, thanks for the example!

My point is that with this solution we have to assume that the `sidebarRendererFn` is in a scope that has access to the same `onChange` method as the editor.

So, everyone using a custom sidebar needs to pass the `onChange` method both to the editor and to the sidebar.

File: src/components/MegadraftEditor.js [hunk 4]
```diff
@@ -162,11 +181,7 @@ export default class MegadraftEditor extends Component {
           className="megadraft-editor"
           id="megadraft-editor"
           ref="editor">
-          <Sidebar
-            plugins={plugins}
-            editorState={editorState}
-            readOnly={this.state.readOnly}
-            onChange={this.onChange} />
+          {this.Sidebar}
```
Comment (ricobl) [inline src/components/MegadraftEditor.js]: I believe that this is a desirable behavior, the `render` method should have control over the sidebar and React does the rest.

We could run into problems of not updating the sidebar.

If it gets slow we'll have to deal with later.

**Comment by SeverS** [comment]
@ricobl I`ve done the suggested changes.
Thank you,

**Comment by ricobl** [comment]
Thanks for the contribution @SeverS!

The `0.4.0` version is released:

https://www.npmjs.com/package/megadraft

It would be awesome to see your custom sidebar, please make it public if you can.

**Pull request merged**